### DISCUSSION TITLE
Feature Request: Enable Function

### DISCUSSION THREAD

**Discussion by Ursus-GIT5805** [minad/tempel#147] [Ideas]
I always use tempel-abbrev-mode and it would be nice to have certain templates be expandable when a certain condition is true.

Then we could do something like this:
```elisp
org-mode :enable-function (org-in-src-block-p)

(main "int main() {" n
  > q n
  "}">)
;; other templates which are enabled on this condition
```
... and it only expands if the cursor is in an org-src-block.
:when only decides whether to load the templates or not.

I've look a bit through the code, one would have to add additional information to each table.
However, the templates are all loaded together so one would have to change the way templates are loaded.
So, I'm asking for the opinions on this (or possible workarounds).

**Comment by minad** [comment]
There is an example for `:when` in https://github.com/minad/tempel?tab=readme-ov-file#template-file-format.

**Reply by Ursus-GIT5805** [reply]
That is clear. However, `:when` only decides whether to load the templates on `tempel-abbrev-mode`. It doesn't use it as as `:enable-function` for the abbrev table, which I would like as a feature.

**Reply by minad** [reply]
Ah, I see. This is not supported I guess. If you want you can propose a patch. See https://github.com/minad/tempel?tab=readme-ov-file#contributions.
